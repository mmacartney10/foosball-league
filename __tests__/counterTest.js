import React from 'react';
import Counter from '../app/counter/counter';
import renderer from 'react-test-renderer';

test('Count increments by one when button clicked', () => {

  const component = renderer.create(
    <Counter/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
