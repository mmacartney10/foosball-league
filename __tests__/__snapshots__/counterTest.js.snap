exports[`test Count increments by one when button clicked 1`] = `
<div>
  <h1>
    0
  </h1>
  <button
    onClick={[Function]}>
    Increment
  </button>
</div>
`;
